### PR TITLE
feat(Anchor): add getCurrentAnchor prop

### PR DESCRIPTION
**Pull request by shaodahong**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#17665 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Customize the anchor highlight      |
| 🇨🇳 Chinese |     自定义锚点高亮      |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/anchor/demo/customizeHighlight.md](https://github.com/shaodahong/ant-design/blob/feat-anchor-getCurrentAnchor/components/anchor/demo/customizeHighlight.md)
[View rendered components/anchor/index.en-US.md](https://github.com/shaodahong/ant-design/blob/feat-anchor-getCurrentAnchor/components/anchor/index.en-US.md)
[View rendered components/anchor/index.zh-CN.md](https://github.com/shaodahong/ant-design/blob/feat-anchor-getCurrentAnchor/components/anchor/index.zh-CN.md)

### CODE REVIEW

**Comment by shaodahong** [comment]
@zombieJ Please check this pr

File: components/anchor/index.en-US.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ For displaying anchor hyperlinks on page and jumping between them.
 | offsetTop | Pixels to offset from top when calculating position of scroll | number | 0 |  |
 | showInkInFixed | Whether show ink-balls in Fixed mode | boolean | false |  |
 | onClick | set the handler to handle `click` event | Function(e: Event, link: Object) | - | 3.9.0 |
+| getCurrentAnchor | Customize the anchor highlight | () => string | - |  |
```
Comment (zombieJ) [inline components/anchor/index.en-US.md]: feature branch with version `3.21.0`

**Comment by zombieJ** [comment]
> feature branch ...

Not master : )

**Comment by shaodahong** [comment]
切成 `feature` 了

File: components/anchor/Anchor.tsx [hunk 2]
```diff
@@ -99,6 +99,7 @@ export interface AnchorProps {
   affix?: boolean;
   showInkInFixed?: boolean;
   getContainer?: () => AnchorContainer;
+  getCurrentAnchor?: () => string;
```
Comment (ztplz) [inline components/anchor/Anchor.tsx]: 加个`jsdoc`吧

**Review by zombieJ — APPROVED** [review]

**Comment by zombieJ** [comment]
Approved. @afc163 @ztplz please help to double confirm.

**Pull request merged**